generators/structure-component/index.ts

```diff
@@ -0,0 +1,99 @@
+import camelCase from "camel-case";
+import Generator from "yeoman-generator";
+import { nameQuestion } from "../../src/questions";
+
+interface FileConfigWithStringTemplate {
+  templateName: string;
+  fileNameSuffix?: string;
+  questions?: any;
+}
+
+interface FileConfigWithFunctionTemplate {
+  templateName: (type: ComponentType) => string;
+  fileNameSuffix: string;
+  questions?: any;
+}
+
+type FileConfig = FileConfigWithStringTemplate | FileConfigWithFunctionTemplate;
+
+enum ComponentType {
+  CLASS = "class",
+  FUNCTION = "function"
+}
+
+export = class StructureComponentGenerator extends Generator {
+  private name: any;
+  private answers: any;
+
+  private fileConfigs: FileConfig[] = [
+    {
+      templateName: (type: ComponentType) => `component.${type}.tsx`,
+      fileNameSuffix: "tsx",
+      questions: {
```

Comment:
to nie powinno być czasem question? albo jeśli questions, to to powinna być tablica :)